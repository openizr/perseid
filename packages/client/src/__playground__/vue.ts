/* c8 ignore start */

import {
  Router,
} from 'scripts/vue/index';
import {
  i18n,
  model,
  store,
  apiClient,
} from '__playground__/services';
import '__playground__/style.scss';
import { type Component, createApp } from 'vue';

function main(): void {
  // Creating Vue root...
  const app = createApp(Router as unknown as Component, {
    container: document.querySelector('#root'),
    services: {
      i18n,
      model,
      store,
      apiClient,
    },
  });
  (app as { mount: (root: string) => void; }).mount('#root');
}

// Ensures DOM is fully loaded before running app's main logic.
// Loading hasn't finished yet...
if (document.readyState === 'loading') {
  document.addEventListener('DOMContentLoaded', main);
  // `DOMContentLoaded` has already fired...
} else {
  main();
}
