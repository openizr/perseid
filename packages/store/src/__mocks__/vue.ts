/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

export const ref = vi.fn(() => ({ value: 'test' }));
export const onMounted = vi.fn((callback: () => unknown) => callback());
export const onUnmounted = vi.fn((callback: () => unknown) => callback());
