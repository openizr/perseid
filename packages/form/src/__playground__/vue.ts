/* c8 ignore start */

import Form from 'scripts/vue/Form.vue';
import Engine from 'scripts/core/Engine';
import Step from 'scripts/vue/DefaultStep.vue';
import { type Component, createApp } from 'vue';

let app: unknown;
const { log } = console;

function main(): void {
  app = createApp(Form as unknown as Component, {
    stepComponent: Step,
    configuration: {
      root: 'root',
      fields: {
        test: { type: 'string', required: true },
      },
      steps: {
        root: {
          fields: ['test'],
        },
      },
      plugins: [
        (engine: Engine): void => {
          engine.on('userAction', (data, next) => {
            log(data);
            return next(data);
          });
        },
      ],
    },
  });
  (app as { mount: (root: string) => void; }).mount('#root');
}

// Ensures DOM is fully loaded before running app's main logic.
// Loading hasn't finished yet...
if (document.readyState === 'loading') {
  document.addEventListener('DOMContentLoaded', main);
  // `DOMContentLoaded` has already fired...
} else {
  main();
}

// Ensures subscriptions to Store are correctly cleared when page is left, to prevent "ghost"
// processing, by manually unmounting Vue components tree.
window.addEventListener('beforeunload', () => {
  (app as { unmount: () => void; }).unmount();
});
