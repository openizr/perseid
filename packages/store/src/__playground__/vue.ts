/* c8 ignore start */

import App from '__playground__/App.vue';
import { Component, createApp } from 'vue';

let app: unknown;

function main(): void {
  app = createApp(App as unknown as Component);
  (app as { mount: (root: string) => void; }).mount('#root');
}

// Ensures DOM is fully loaded before running app's main logic.
// Loading hasn't finished yet...
if (document.readyState === 'loading') {
  document.addEventListener('DOMContentLoaded', main);
  // `DOMContentLoaded` has already fired...
} else {
  main();
}

// Ensures subscriptions to Store are correctly cleared when page is left, to prevent "ghost"
// processing, by manually unmounting Vue components tree.
window.addEventListener('beforeunload', () => {
  (app as { unmount: () => void; }).unmount();
});
