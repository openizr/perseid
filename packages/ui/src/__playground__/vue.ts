/* c8 ignore start */

import { createApp, type App } from 'vue';
import AppRouter from '__playground__/pages/AppRouter.vue';

let app: App;

function main(): void {
  app = createApp(AppRouter as unknown as App);
  app.mount('#root');
}

// Ensures DOM is fully loaded before running app's main logic.
// Loading hasn't finished yet...
if (document.readyState === 'loading') {
  document.addEventListener('DOMContentLoaded', main);
  // `DOMContentLoaded` has already fired...
} else {
  main();
}

// Ensures subscriptions to Store are correctly cleared when page is left, to prevent "ghost"
// processing, by manually unmounting Vue components tree.
window.addEventListener('beforeunload', () => {
  app.unmount();
});
