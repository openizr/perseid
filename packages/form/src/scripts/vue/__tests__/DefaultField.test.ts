/**
 * Copyright (c) Openizr. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @vitest-environment jsdom
 */

import { type DefineComponent } from 'vue';
import type Engine from 'scripts/core/Engine';
import { render } from '@testing-library/vue';
import DefaultField from 'scripts/vue/DefaultField.vue';

describe('vue/DefaultField', () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  test('renders correctly - non-binary field', () => {
    const { container } = render(DefaultField, {
      props: {
        error: null,
        value: null,
        active: false,
        required: true,
        type: 'string',
        status: 'initial',
        fields: undefined,
        path: 'root.0.test',
        useSubscription: vi.fn(),
        engine: {} as unknown as Engine,
        field: {} as unknown as DefineComponent,
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  test('renders correctly - binary field', () => {
    const { container } = render(DefaultField, {
      props: {
        error: null,
        value: null,
        active: false,
        required: true,
        type: 'binary',
        status: 'initial',
        fields: undefined,
        path: 'root.0.test',
        useSubscription: vi.fn(),
        engine: {} as unknown as Engine,
        field: {} as unknown as DefineComponent,
      },
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
