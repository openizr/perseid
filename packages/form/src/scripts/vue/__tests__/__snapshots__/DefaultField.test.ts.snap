// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vue/DefaultField > renders correctly - binary field 1`] = `
<pre
  field="[object Object]"
>
      {
  "path": "root.0.test",
  "type": "binary",
  "status": "initial",
  "error": null,
  "active": false,
  "required": true,
  "value": "&lt;Binary&gt;",
  "field": "&lt;Component&gt;",
  "engine": "&lt;Engine&gt;",
  "useSubscription": "&lt;Function&gt;"
}
  
</pre>
`;

exports[`vue/DefaultField > renders correctly - non-binary field 1`] = `
<pre
  field="[object Object]"
>
      {
  "path": "root.0.test",
  "type": "string",
  "status": "initial",
  "error": null,
  "active": false,
  "required": true,
  "value": null,
  "field": "&lt;Component&gt;",
  "engine": "&lt;Engine&gt;",
  "useSubscription": "&lt;Function&gt;"
}
  
</pre>
`;
